sky/clouds/azure.py

```diff
@@ -250,13 +250,13 @@ def get_feasible_launchable_resources(self, resources):
 
         def failover_disk_tier(instance_type: str,
                                disk_tier: Optional[str]) -> Optional[str]:
-            if disk_tier is None:
-                ok, _ = Azure.check_disk_tier(instance_type,
-                                              clouds.Cloud._DEFAULT_DISK_TIER)
-                if not ok:
-                    # Auto failover to low disk tier when disk tier
-                    # are not specified
-                    return 'low'
+            if disk_tier:
+                ok, _ = Azure.check_disk_tier(instance_type, disk_tier)
+                return disk_tier if ok else None
+            disk_tier = clouds.Cloud._DEFAULT_DISK_TIER
+            all_tiers = ['high', 'medium', 'low']
+            while not Azure.check_disk_tier(instance_type, disk_tier)[0]:
+                disk_tier = all_tiers[all_tiers.index(disk_tier) + 1]
```

Comment:
Since the `_DEFAULT_DISK_TIER` is set to `medium`, we are not able to check the `high` disk_tier in this loop right?

Should we instead do the following:
```suggestion
            all_tiers = {'high', 'medium', 'low'}
            while not Azure.check_disk_tier(instance_type, disk_tier)[0]:
                all_tiers.remove(disk_tier)
                if not all_tiers:
                    # No available disk_tier found the specified instance_type
                    return None
                disk_tier = list(all_tiers)[0]
```